rest/handler.go

```diff
@@ -468,9 +470,11 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 		if err != nil {
 			return err
 		}
-		h.collection, err = h.db.GetDatabaseCollectionWithUser(*keyspaceScope, *keyspaceCollection)
-		if err != nil {
-			return err
+		if ks != "" {
+			h.collection, err = h.db.GetDatabaseCollectionWithUser(*keyspaceScope, *keyspaceCollection)
```

Comment:
I'm a bit worried it is dangerous to not always set a a collection, because we never check later to see if `h.collection` is nil or not. We also never check to see if `h.db` is present so this is probably fine.